scripts/nfd-must-gather.sh

```
@@ -0,0 +1,393 @@
+    # Check default NFD namespace
+    if oc get namespace "${NFD_NAMESPACE}" &> /dev/null; then
+        nfd_namespaces+=("${NFD_NAMESPACE}")
+    fi
+
+    # Look for NFD resources in all namespaces
+    log_info "Searching for NFD resources across all namespaces"
+
+    # Find namespaces with NFD pods
+    while IFS= read -r ns; do
+        if [[ -n "$ns" && "$ns" != "NAMESPACE" ]]; then
+            if [[ ! " ${nfd_namespaces[*]} " =~ " ${ns} " ]]; then
```

Shellcheck (0.10.0)

[warning] Remove quotes from right-hand side of =~ to match as a regex rather than literally.

(SC2076)